From 10b4d928dfd933ad2fac935a52b638cfe07835f6 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Fri, 16 Jan 2026 00:07:52 -0700
Subject: [PATCH 50/76] media: ipu6: isys: Use
 v4l2_ctrl_subdev_subscribe_event()

The ipu6-isys driver uses v4l2_ctrl_subscribe_event() to handle control
event subscription on a subdev. While this works, it is the wrong API.
Use the subdev-specific v4l2_ctrl_subdev_subscribe_event() helper
instead.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
Signed-off-by: Hans Verkuil <hverkuil+cisco@kernel.org>
---
 6.18.0/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/6.18.0/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c b/6.18.0/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
index 6030bd2..d1fece6 100644
--- a/6.18.0/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
+++ b/6.18.0/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
@@ -109,7 +109,7 @@ static int csi2_subscribe_event(struct v4l2_subdev *sd, struct v4l2_fh *fh,
 	case V4L2_EVENT_FRAME_SYNC:
 		return v4l2_event_subscribe(fh, sub, 10, NULL);
 	case V4L2_EVENT_CTRL:
-		return v4l2_ctrl_subscribe_event(fh, sub);
+		return v4l2_ctrl_subdev_subscribe_event(sd, fh, sub);
 	default:
 		return -EINVAL;
 	}
-- 
2.43.0

